### DISCUSSION TITLE
Dog bed functionality: Respawn a dog

### DISCUSSION THREAD

**Discussion by MrKacafirekCZ** [MrKacafirekCZ/doggo-mod-overhauled#32] [Polls]
I'm considering adding a functionality to the dog bed and I would like some opinions. A feature suggestion (#29) has been made to add a way for dogs to respawn.

**Comment by rheena211** [comment]
I came here to ask about removing dog beds entirely since I use the Domestic Innovations mod. Is there any way these two mods could be compatible?

**Comment by theavandiepen** [comment]
I would like them to lie down in their beds sometimes (maybe when the player goes to sleep? maybe also sometimes during the day if near the bed?). That would be super cute

**Reply by Some-Doofus** [reply]
I'd also like to see these features added, and to add to them there should be a way to tell the dog to go home without just leaving it sitting the whole time. The bed would serve as the dogs home and it would be free to wander around a certain radius of the bed and return to it at night to sleep. If you wanted even more functionality you could train it to be a guard dog and guard around its bed during the day and/or night. Giving it a food bowl would allow it to regenerate health on its own when needed.

**Reply by MrKacafirekCZ** [reply]
> there should be a way to tell the dog to go home without just leaving it sitting the whole time

I've been thinking about how to do specific commands, maybe even tricks, to a specific Doggo however I haven't really found a good solution as to how a player would execute the huge potential amount of commands/tricks to a specific Doggo:
- Keybinds/key shortcuts? Already you have Right-Click (Makes the Doggo sit), Right-Click + Food in hand (Gives the food to the Doggo) and Shift + Right-Click + Food in hand (Breeding). Adding more is just gonna make it unnecessarily complicated, let alone remembering all the possible shortcuts that you as the player can do.
- Custom items, each with specific command/trick? That wouldn't really be intuitive + you would end up with a full inventory of just Doggo item commands/tricks. And the amount of crafting recipes there would be.
- One item, where you would scroll to select which command/trick to do? Again, once there are going to be many and many commands/tricks, you will end up with bad game design mechanic. How is it going to take to find that one very specific command you're looking for?

> The bed would serve as the dogs home and it would be free to wander around a certain radius of the bed and return to it at night to sleep. If you wanted even more functionality you could train it to be a guard dog and guard around its bed during the day and/or night. Giving it a food bowl would allow it to regenerate health on its own when needed.

Definitely a cool concept. I like it. But it would be a big hit into Doggo's AI, basically stripping out almost the entire original AI and replacing it with a new one.
